vars/abapEnvironmentPipelineStageATC.groovy

```diff
@@ -0,0 +1,28 @@
+import com.cloudbees.groovy.cps.NonCPS
+import com.sap.piper.GenerateStageDocumentation
+import groovy.transform.Field
+
+import static com.sap.piper.Prerequisites.checkScript
+
+@Field String STEP_NAME = getClass().getName()
+@Field Set GENERAL_CONFIG_KEYS = []
+@Field STAGE_STEP_KEYS = []
+@Field Set STEP_CONFIG_KEYS = []
+@Field Set PARAMETER_KEYS = []
+/**
+ * This stage runs the ATC Checks
+ */
+void call(Map parameters = [:]) {
+    def script = checkScript(this, parameters) ?: this
+
+    def stageName = parameters.stageName?:env.STAGE_NAME
+    piperStageWrapper (script: script, stageName: stageName) {
+
+        abapEnvironmentRunATCCheck script: parameters.script
+
+        def atcResult = readFile file: "ATCResults.xml"
+        if (atcResult != '<?xml version="1.0" encoding="utf-8"?><checkstyle version="1.0"/>') {
+            unstable('ATC Issues detected - setting build status to UNSTABLE')
```

Comment:
shouldn't it be possible to break the pipeline in case of ATC check errors?

Reply:
yes, I'm currently working on changing the behavior. The checkstyle plugin should be used via a stage extension. When trying to use the stage Wrapper for all stages, I am facing the following error without further details:
```
java.lang.NoSuchMethodError: No such DSL method 'unstashStageFiles' found among steps [...]
00:54:56.809  	at org.jenkinsci.plugins.workflow.cps.DSL.invokeMethod(DSL.java:202)
00:54:56.809  	at org.jenkinsci.plugins.workflow.cps.CpsScript.invokeMethod(CpsScript.java:122)
00:54:56.809  	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:48)
00:54:56.809  	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
00:54:56.809  	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
00:54:56.809  	at com.cloudbees.groovy.cps.sandbox.DefaultInvoker.methodCall(DefaultInvoker.java:20)
00:54:56.809  	at piperStageWrapper.executeStage(piperStageWrapper.groovy:65)
00:54:56.809  	at piperStageWrapper.call(piperStageWrapper.groovy:41)
```
The error occurs in the stage "Prepare System". I guess I'm missing some details. Do you have any suggestions?

Reply:
The reason seems to be that config.stashContent was not filled properly.
Then `utils.unstashStageFiles(script, stageName, config.stashContent)` fails.
However, the default was not considered
`def unstashStageFiles(Script script, String stageName, List stashContent = [])`